### PR TITLE
Add docs Project on The Default Branch

### PR DESCRIPTION
**Pull request by khajavi**

### CODE REVIEW

**Review by sideeffffect — DISMISSED** [review]

**Comment by sideeffffect** [comment]
Red build

File: .github/workflows/site.yml [hunk 1]
```diff
@@ -1,35 +1,98 @@
...
+        java-version: 17
+        check-latest: true
+    - name: Check if the README file is up to date
+      run: sbt  docs/checkReadme
+    - name: Check if the site workflow is up to date
+      run: sbt  docs/checkGithubWorkflow
+    - name: Check artifacts build process
+      run: sbt  +publishLocal
+    - name: Check website build process
+      run: sbt docs/clean; sbt  docs/buildWebsite
```
Comment (sideeffffect) [inline .github/workflows/site.yml:35]:
```
docs/clean
```
Is this really needed? Clean after what?

Comment (khajavi) [reply .github/workflows/site.yml:35]: Yes, we need that. The `docs/buildWebsite` needs a clean state.

**Comment by khajavi** [comment]
@sideeffffect 
Can you help me on making `docs` compatible with this branch? There are some errors on compiling docs: https://github.com/zio/zio-prelude/actions/runs/3931431977/jobs/6722668537#step:4:56

**Comment by sideeffffect** [comment]
Just for experiment, what happens if you remove the `mdoc:fail` from the `Equal[String => Int]` block?

File: build.sbt [hunk 2]
```diff
@@ -232,27 +232,23 @@ lazy val benchmarks = project
...
+    crossScalaVersions -= Scala211,
+    projectName                                := "ZIO Prelude",
+    mainModuleName                             := (core.jvm / moduleName).value,
+    projectStage                               := ProjectStage.ProductionReady,
     ScalaUnidoc / unidoc / unidocProjectFilter := inProjects(
-      core.jvm,
-      experimental.jvm,
-      experimentalLaws.jvm,
-      laws.jvm,
-      scalaParallelCollections
```
Comment (sideeffffect) [inline build.sbt]: Why remove these?

**Comment by sideeffffect** [comment]
Problem in `These.scala:156:34`, most likely in Scala 2.11.
Perhaps more type annotations would help here?

File: build.sbt (hunk 2 above)
Comment (khajavi) [reply build.sbt]: I'm making some experimental changes to make ci passes. Finally, I can revert unrelated changes.

Comment (sideeffffect) [reply build.sbt]: Good luck :crossed_fingers:

**Review by sideeffffect — APPROVED** [review]
:heart: thank you so much

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**